source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -156,6 +156,39 @@ Filters are useful when you need an immediate response that directly modifies th
+| **Purpose**                                                        | Allow extending functionality of the Open edX platform without modifying the                                  | Modify the code directly to meet specific needs.                                                                                   |
+|                                                                    | core codebase but by leveraging plugins and configurations.                                                   |                                                                                                                                    |
++--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| **Advantages**                                                     | - More decoupled implementations.                                                                             | - Full control of the platform behavior.                                                                                           |
+|                                                                    | - Decreased maintenance burden of the platform, maintenance efforts are centered on plugins.                  | - No restrictions from what can be changed.                                                                                        |
+|                                                                    | - Enables reusable and configurable behavior across the platform.                                             | - Suitable for highly customized use cases.                                                                                        |
+|                                                                    | - Easier to maintain long-term due to backward compatibility and versioning policies.                         |                                                                                                                                    |
++--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| **Disadvantages**                                                  | - Can increase complexity in debugging and maintenance because it adds more processing layers.                | - Separation from the core makes long-term maintenance challenging.                                                                |
```

Comment:
I think we should call it "separation from the upstream project" because core has a different meaning here.

Reply:
That's more clear. Thank you.

https://github.com/openedx/docs.openedx.org/pull/617/commits/67bd6f7c4060421d7b2dea42c6affffa8a8d6acd#diff-01d0bcd5d9113aaa8a7f3fde94c7e5ca100265eebebbad867dbf4de21123d451R177